lib/write_concern.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+/**
+ * The **WriteConcern** class is a class that represents a MongoDB WriteConcern.
+ * @class
+ * @param {} w
+ * @param {} wtimeout
+ * @param {} j
+ * @param {} fsync
+ * @return {WriteConcern}
+ */
+class WriteConcern {
+  constructor(w, wtimeout, j, fsync) {
+    if (w != null) {
+      this.w = w;
+    }
+    if (wtimeout != null) {
+      this.wtimeout = wtimeout;
+    }
+    if (j != null) {
+      this.j = j;
+    }
+    if (fsync != null) {
+      this.fsync = fsync;
+    }
+  }
+
+  isDefined() {
```

Comment:
I wonder if we can avoid having this method by instead providing a static `fromOptions` method? Something like
```
class WriteConcern {
    static fromOptions(options) {
        if (options.w == null && options.wtimeout == null && options.j == null && options.fsync == null) {
          return null;
        }
   
        // extract the values and create a `WriteConcern` and return it
    }
}
```

This way above you could always store a `writeConcern` property in `this.s`, and in the getters just return `this.s.writeConcern` instead of creating a new one and checking if its valid every time.